### PR TITLE
feat(check): Add `--allow-abort` option

### LINKED ISSUE #424: Aborted Commit Considered to Fail Validation When Run As commit-msg Hook
## Description

Git interprets an empty commit message to mean that a commit should be aborted, but calls all commit-msg hooks anyways. This causes pre-commit hooks that run as commit-msg hooks but don't guard against empty commit messages to spew errors en masse when the commit was already being intentionally aborted.

## Steps to reproduce
1. Configure commitizen to run as a commit-msg hook using pre-commit as described below.
2. `touch foo && git add && git commit`
3. Save an empty commit message.

For step 1, add the following to `.pre-commit-config.yaml`:

```
yaml
  - repo: https://github.com/commitizen-tools/commitizen
    rev: v2.18.1
    hooks:
      - id: commitizen
        stages:
          - commit-msg
```

## Current behavior

[Pre-commit follows git's lead and calls commitizen when the commit message is empty when run as a commit-msg hook](https://github.com/pre-commit/pre-commit/issues/2052). This leads to the following error:

```
commitizen check..............................................................Failed
- hook id: commitizen
- exit code: 14

commit validation: failed!
please enter a commit message in the commitizen format.
commit "": "# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch foo
#
# Changes to be committed:
#       ,,,
pattern: (build|ci|docs|feat|fix|perf|refactor|style|test|chore|revert|bump)(\(\S+\))?!?:(\s.*)
```

The commit is still correctly aborted of course.

## Desired behavior

I expected the commitizen commit-msg hook to silently report success in this case since there is no commit message to check.

## Environment

- commitizen version: 2.18.1
- python version: Python 3.9.5
- operating system: Linux

### PR DESCRIPTION
**Pull request by Kurt-von-Laven**
## Description

Empty commit messages indicate to Git that the commit should be aborted. Displaying an error message when the commit is already being aborted typically only creates confusion. Add an `--allow-abort` argument to the check command and `allow_abort` config variable, both defaulting to false for backwards compatibility. When the commit message is empty, determine the outcome based on the `allow_abort` setting alone, ignoring the pattern.

Closes #424.

## Checklist

- [x] Add test cases to all the changes you introduce
- [x] Run `./scripts/format` and `./scripts/test` locally to ensure this change passes linter check and test
- [x] Test the changes on the local machine manually
- [x] Update the documentation for the changes

## Expected behavior

No error message is displayed when aborting a commit locally, unless the commit .

## Steps to Test This Pull Request

1. Run `cz commit`.
2. Save and close the commit message file that appears with an empty commit message.
3. No error occurs if you add `allow_abort = true` to your config, but an error occurs otherwise.

## Additional context

Would it make sense for `--allow-abort` to also be added as an arg to the `commit` command? How do folks feel about the argument/setting name and overall approach? See also #247 for a related feature proposal. Since the time of that discussion, I realized that naming a Commitizen argument `--allow-empty-message` could prove extremely confusing, because passing it to Commitizen would not, by itself, cause commits with empty messages to be committed like the Git argument of the same name. It would also interact poorly with the proposal to allow ferrying arbitrary arguments to `git commit` in #451, because it would be easy to accidentally pass the argument to Git, resulting in commits with empty messages no longer being aborted.

### CODE REVIEW

**Comment by Woile** [comment]
LGTM but `cz check` is not my speciality, can u take a look @Lee-W ?

File: commitizen/cli.py [hunk 1]
```diff
@@ -240,6 +240,12 @@
                         "help": "commit message that needs to be checked",
                         "exclusive_group": "group1",
                     },
+                    {
+                        "name": ["--reject-abort"],
+                        "action": "store_true",
+                        "default": False,
```
Comment (Lee-W) [inline commitizen/cli.py:246]: If I'm not misunderstood, the original behavior is `True`, right? If that's the case, I'd prefer to use that as the default option, unless we're to bump our major version

File: docs/check.md [hunk 2]
```diff
@@ -46,3 +50,12 @@ $ cz check --commit-msg-file COMMIT_MSG_FILE
 
 In this option, COMMIT_MSG_FILE is the path of the temporal file that contains the commit message.
 This argument can be useful when cooperating with git hook, please check [Automatically check message before commit](auto_check.md) for more information about how to use this argument with git hook.
+
+### Reject Abort
+
```
Comment (Lee-W) [inline docs/check.md:55]: nitpick: maybe we could remove this additional blank line

File: commitizen/commands/check.py [hunk 3]
```diff
@@ -79,15 +83,15 @@ def __call__(self):
 
     def _get_commits(self):
         # Get commit message from file (--commit-msg-file)
-        if self.commit_msg_file:
+        if self.commit_msg_file is not None:
```
Comment (Lee-W) [inline commitizen/commands/check.py:86]: Wondering why you prefer to use `is not None` in this case?

File: commitizen/commands/check.py [hunk 4]
```diff
@@ -79,15 +83,15 @@ def __call__(self):
...
             with open(self.commit_msg_file, "r", encoding="utf-8") as commit_file:
                 msg = commit_file.read()
             msg = self._filter_comments(msg)
             msg = msg.lstrip("\n")
             commit_title = msg.split("\n")[0]
             commit_body = "\n".join(msg.split("\n")[1:])
             return [git.GitCommit(rev="", title=commit_title, body=commit_body)]
-        elif self.commit_msg:
+        elif self.commit_msg is not None:
```
Comment (Lee-W) [inline commitizen/commands/check.py:94]: Same here

**Review by Lee-W — COMMENTED** [review]
Hi @Kurt-von-Laven , sorry for late reviewing. Your work is fantastic! Just left a few questions. But I think we're almost ready to mrege

**Review by Kurt-von-Laven — COMMENTED** [review]
Would it make sense for `--allow-abort` to also be added as an arg to the commit command?

EDIT: I think I answered this question for myself: no.

**Comment by Kurt-von-Laven** [comment]
I am not able to reproduce the failure on Python 3.10 locally on Python 3.10.4. It appears to be unrelated to this pull request, so I think the build may simply need to be re-run. Please let me know if anyone sees a connection I am missing.

File: commitizen/cli.py (hunk 1 above)
Comment (Lee-W) [reply commitizen/cli.py:246]: Sounds good 👍

File: commitizen/commands/check.py (hunk 3 above)
Comment (Lee-W) [reply commitizen/commands/check.py:86]: Got it 👍

File: commitizen/commands/check.py (hunk 4 above)
Comment (Lee-W) [reply commitizen/commands/check.py:94]: In that case, could you please add a comment here. I think we might miss it in the future

**Review by Lee-W — APPROVED** [review]
Except for the minor comment update, this PR looks good to me. @Woile I'm planning on merging it this weekend if no other new comments added

**Comment by Lee-W** [comment]
hmmm. we might need to take a deeper look at the 3.10 case

File: commitizen/commands/check.py (hunk 4 above)
Comment (Kurt-von-Laven) [reply commitizen/commands/check.py:94]: Good idea; done. Also, removing `is not None` on this line (but not the one above) will currently cause tests to fail.

**Review by Kurt-von-Laven — COMMENTED** [review]
I am still unable to reproduce the build failure locally on Python 3.10.4.

**Comment by Kurt-von-Laven** [comment]
Looks like the Python 3.10 build indeed simply needed to be re-run since all I changed was the two comments.

**Review by Lee-W — APPROVED** [review]
LGTM. Let's merge it now!

**Pull request merged**